caas/kubernetes/provider/k8s.go

```diff
@@ -409,11 +409,11 @@ func (k *kubernetesClient) Create(envcontext.ProviderCallContext, environs.Creat
 
 // Bootstrap deploys controller with mongoDB together into k8s cluster.
 func (k *kubernetesClient) Bootstrap(
-	ctx environs.BootstrapContext,
+	ctx context.Context,
+	cmdCtx environs.BootstrapContext,
 	callCtx envcontext.ProviderCallContext,
```

Comment:
I think BoostrapContext should be extended to gain a Context() method to return the context.Context.
It already has 
```
	GetStdin() io.Reader
	GetStdout() io.Writer
	GetStderr() io.Writer
	Infof(format string, params ...interface{})
	Verbosef(format string, params ...interface{})
...
```
so we'd be adding to that. Also reduces the churn in method signatures.

Reply:
Done in https://github.com/juju/juju/pull/12475